force_bdss/mco/optimizers/scipy_optimizer.py

```diff
@@ -0,0 +1,265 @@
+import numpy as np
+from functools import partial
+from traits.api import (
+    Enum,
+    provides,
+    HasStrictTraits
+)
+
+from force_bdss.mco.parameters.mco_parameters import (
+    RangedMCOParameter,
+    RangedVectorMCOParameter
+)
+
+from force_bdss.mco.optimizers.i_optimizer import IOptimizer
+
+from scipy import optimize as scipy_optimize
+
+
+class ScipyTypeError(Exception):
+    pass
+
+
+@provides(IOptimizer)
+class ScipyOptimizer(HasStrictTraits):
+    """ Optimization of an objective function using scipy.
+    """
+
+    #: Algorithms available to work with
+    algorithms = Enum("SLSQP", "Nelder-Mead", "Powell", "CG", "BFGS",
+                      "Newton-CG", "L-BFGS-B", "TNC", "COBYLA",
+                      "trust-constr", "dogleg",
+                      "trust-ncg", "trust-exact", "trust-krylov")
+
+    def optimize_function(self, func, params):
+        """ Minimize the passed function.
+
+        Parameters
+        ----------
+        func: Callable
+            The MCO function to optimize
+            Takes a list of MCO parameter values.
```

Comment:
Maybe include 'must return a scalar value', if we only support functions that can be optimised by `scipy.optimize.minimize`. Alternatively we could state that any non-scalar value will be converted using a summation (see next comment).